solr/modules/extraction/build.gradle

```diff
@@ -26,55 +26,18 @@ dependencies {
   implementation 'org.apache.lucene:lucene-core'
   implementation 'org.slf4j:slf4j-api'
 
-  // We export tika because other modules depend on it (and its submodules)
```

Comment:
I'm glad this was reversed (api -> implementation); maybe the comment was wrong/obsolete

Reply:
Yea I'm not sure about the comment. I looked into it some and not sure where the extraction module would be referenced.

Reply:
the langid module uses Tika, it's readme says 
> The Tika detector depends on Tika Core (which is part of the extraction module)

But it also puts tika-core in its own `lib/`, so this may be old information? 
Did we duplicate tika-core.jar in 8.x?

Reply:
I checked this and tika-core is not included in the langid lib dir.

Reply:
So I looked into this more and it doesn't seem like there is clear documentation on how this worked in Solr 8.x. It looks like there is an implicit assumption that both `extraction` and `langid` would be on the Solr classpath for `langid` to work with Tika. 

Personally it feels way cleaner to me for each `module` to be independent and therefore be included as standalone. This would end up with duplicate jars. I think this is how packages would work though.

I understand the concern about not having duplicate jars in the distribution due to size though. If we are that concerned about distribution size, wouldn't it make more sense to have each contrib as a separate download instead of trying to play tricks with these specific jars?

I played around with `api` vs `implementation` for this change - and it has no effect on the `langid` module lib. The only way to remove the jar from `langid` is to switch to `compileOnly` instead of `implementation` in the `langid` module. I can make that change if that would be the preferred approach.

Reply:
I don't think the Jar duplication is a big problem for us to make modules non-independent.

Reply:
So I'm on the fence about this with Tika. Since `tika-core` is NOT the only thing needed to do language detection as far as I know. It would be a lot more duplication of jars just to get language detection working. (I'm not even 100% sure that all the tika parsers in the extraction module are right for Tika to do language detection with the langid module). I think leaving it like this is going to be the most expected for now. I do think this needs more thought separately though.

Reply:
Ok; makes sense for now.  Hopefully this is documented for our users.

Reply:
`tika-core` should contain everything needed for language identification. It is 718kb, so not a big deal.
If we don't care for windows users we could symlink that jar from extraction to langid in the tarball :)

Reply:
There is an entire `tika-langdetect` dependency - https://search.maven.org/artifact/org.apache.tika/tika-langdetect/1.28.1/bundle or https://mvnrepository.com/artifact/org.apache.tika/tika-langdetect/1.28.1. I played with `langid` years ago and remember having to add jars to get it to actually work. I could add `tika-langdetect` as `runtimeOnly` in the `langid` module, but that adds more jars.

Reply:
Yea, I think Tika 1.x has the original tika detector in tika-core, while 2.x have moved it out to a separate tika-langdetect. Solr still depends on 1.x. Solr also has a choice of detectors (https://github.com/apache/solr/tree/main/solr/modules/langid/src/java/org/apache/solr/update/processor) which is overlapping with Tika's own (https://github.com/apache/tika/tree/main/tika-langdetect) with our own abstraction. Perhaps it would be better to delegate everything to tika-langdetect and kill our own custom ones. We'd still duplicate tika-core, but our langid module would use tika-langdetect which would not necessarily be needed in extraction module (unless we want to detect language during extraction).

Reply:
Thanks to Jan for looking into this in SOLR-16010 - I reverted the `langid` changes since what is on `main` works and only duplicates `tika-core` jar as expected.